src/qililab/instruments/quantum_machines/quantum_machines_cluster.py

```diff
@@ -282,34 +285,112 @@ def reset(self):
             self._qm.close()
+            self._is_connected_to_qm = False
 
-    def update_configuration(self, compilation_config: dict):
+    def append_configuration(self, configuration: dict):
         """Update the configuration dictionary by appending the configuration generated by compilation."""
-        self._config = merge_dictionaries(dict(self._config), compilation_config)
-        # If we are already connected, reopen the connection with the new configuration
-        if self._is_connected_to_qm():
-            self._qm = self._qmm.open_qm(config=self._config, close_other_machines=True)
+        merged_configuration = merge_dictionaries(dict(self._config), configuration)
+        if self._config != merged_configuration:
+            self._config = cast(DictQuaConfig, merged_configuration)
+            # If we are already connected, reopen the connection with the new configuration
+            if self._is_connected_to_qm:
+                self._qm = self._qmm.open_qm(config=self._config, close_other_machines=True)
 
     def run_octave_calibration(self):
         """Run calibration procedure for the buses with octaves, if any."""
         elements = [element for element in self._config["elements"] if "RF_inputs" in self._config["elements"][element]]
         for element in elements:
             self._qm.calibrate_element(element)
 
-    def set_parameter(self, parameter: Parameter, value: float | str | bool, channel_id: int | None = None):
-        """Sets the parameter of a specific instrument.
+    def set_parameter_of_bus(self, bus: str, parameter: Parameter, value: float | str | bool):
+        """Sets the parameter of the instrument
 
         Args:
-            parameter (Parameter): parameter settings of the instrument to update
-            value (float | str | bool): value to update
-            channel_id (int | None, optional): instrument channel to update, if multiple. Defaults to None.
+            bus (str): The assossiated bus to change parameter
+            parameter (Parameter): The parameter to update
+            value (float | str | bool): The new value of the parameter
 
-        Returns:
-            bool: True if the parameter is set correctly, False otherwise
+        Raises:
+            NotImplementedError: Raised if not connected to Quantum Machines
+            ParameterNotFound: Raised if parameter does not exist
+        """
+        # TODO: Change private QM API to public when implemented.
+        if not self._is_connected_to_qm:
+            raise NotImplementedError(f"You should be connected to {self.name} in order to change a parameter.")
+
+        element = next((element for element in self.settings.elements if element["bus"] == bus), None)
+        if element is None:
+            raise ValueError(f"Bus {bus} was not found in {self.name} settings.")
+
+        if parameter in [Parameter.LO_FREQUENCY, Parameter.GAIN]:
+            if "rf_inputs" not in element:
```

Comment:
is `GAIN` controlled in the LO (i guess that's the octave) for QM? just curious

Reply:
Yes, it's this one: https://docs.quantum-machines.co/1.1.6/qm-qua-sdk/docs/API_references/qm_octave/#qm.octave.qm_octave.QmOctave.set_rf_output_gain